lib/client.js

```diff
@@ -212,6 +212,7 @@ Client.prototype._setSequenceArgs = function(argsScheme, args) {
     return args;
   }
   for (var partIndex in argsScheme) {
+    partIndex = partIndex.replace('[]','')
```

Comment:
This seems like a pretty big breaking change to me.  Just because there aren't tests, doesn't mean it's good for use.  What are we trying to solve here?